src/main/java/org/jenkinsci/plugins/oic/ssl/IgnoringHostNameVerifier.java

```diff
@@ -0,0 +1,25 @@
+package org.jenkinsci.plugins.oic.ssl;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSession;
+import jenkins.security.FIPS140;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * {@link HostnameVerifier} that accepts any presented hostanme including incorrect ones.
+ */
+@Restricted(NoExternalUse.class)
+public final class IgnoringHostNameVerifier implements HostnameVerifier {
```

Comment:
Again, enough plugins use something like this that it should be offered from Jenkins core.